android/src/main/java/com/swmansion/rnscreens/NoOpFragment.kt

```diff
@@ -0,0 +1,27 @@
+package com.swmansion.rnscreens.gamma.helpers
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+
+class NoOpFragment: Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Immediately remove the fragment when it's created, to prevent Android system from trying to
+        // restore it later, which can cause crashes on configuration changes.
+        // Fixes: https://github.com/software-mansion/react-native-screens/issues/17
```

Comment:
This is not 100% accurate, let me rephrase that

```suggestion
        // This is the first moment where we have access to non-null parent fragment manager, 
        // so that we can remove the fragment from the hierarchy. 
```

I'd move the rest of the comment to the top of the file, as I've pointed out in another suggestion.